modules/nextflow/src/main/groovy/nextflow/k8s/client/ConfigDiscovery.groovy

```diff
@@ -34,27 +34,20 @@ import org.yaml.snakeyaml.Yaml
 @Slf4j
 class ConfigDiscovery {
 
-    private Map<String,String> env = System.getenv()
-
-    private ClientConfig config
-
-    private String context
-
     /**
      * Discover Kubernetes service from current environment settings
      */
-    ClientConfig discover() {
-
-        config = new ClientConfig()
+    static ClientConfig discover(String context=null) {
```

Comment:
not sure this was a good idea, static methods should be avoided as much as possible because it makes hard testing (between the other things)

Reply:
Okay, I changed ConfigDiscovery to a singleton to make testing easier